File: frontend/src/modules/fleet/types/driver.ts
export interface Driver {
  id: string;
  driverId: string;
  name: string;
  email: string;
  contactNumber: string;
  gender: string;
  department: string;
  licenceNumber: string;
  licenceIssued: string | null;
  licenceExpiry: string | null;
  firstName: string;
  lastName: string;
  workEmail: string;
  workPhone: string;
  whatsappNumber: string | null;
  idNumber: string;
  licenceCode: string | null;
  hasPdp: boolean;
  unitName: string | null;
  streetNumber: string | null;
  streetName: string | null;
  suburb: string | null;
  city: string | null;
  province: string | null;
  postalCode: string | null;
  employeeNumber: string | null;
  jobTitle: string | null;
  branch: string | null;
  employmentType: string;
  employmentStatus: string;
  workStartDate: string | null;
  manager: string | null;
  createdAt?: string;
  createdByUserId?: string | null;
  createdByDisplayName?: string | null;
  updatedAt?: string | null;
  updatedByUserId?: string | null;
  updatedByDisplayName?: string | null;
  linkedUser?: LinkedUser | null;
}

export interface LinkedUser {
  userId: string;
  username: string;
  email: string;
  displayName: string | null;
  isActive: boolean;
}

export interface DriversResponse {
  items: Driver[];
  total: number;
}

export interface SaveDriverRequest {
  firstName: string;
  lastName: string;
  workEmail: string;
  workPhone: string;
  whatsappNumber?: string;
  gender?: string;
  licenceNumber: string;
  idNumber: string;
  licenceIssued?: string | null;
  licenceExpiry?: string | null;
  licenceCode?: string;
  hasPdp: boolean;
  unitName?: string;
  streetNumber?: string;
  streetName?: string;
  suburb?: string;
  city?: string;
  province?: string;
  postalCode?: string;
  employeeNumber?: string;
  jobTitle?: string;
  department?: string;
  branch?: string;
  employmentType?: string;
  employmentStatus?: string;
  workStartDate?: string | null;
  manager?: string;
}

File: Modules/CarTrack.Modules.Fleet/Domain/Driver.cs
using CarTrack.Core;

namespace CarTrack.Modules.Fleet;

public class Driver : IAuditable
{
    public Guid Id { get; set; }

    public required string DriverCode { get; set; }

    public required string FirstName { get; set; }

    public required string LastName { get; set; }

    public required string WorkEmail { get; set; }

    public required string WorkPhone { get; set; }

    public string? WhatsappNumber { get; set; }

    public string Gender { get; set; } = "Other";

    public required string LicenceNumber { get; set; }

    public required string IdNumber { get; set; }

    public DateOnly? LicenceIssued { get; set; }

    public DateOnly? LicenceExpiry { get; set; }

    public string? LicenceCode { get; set; }

    public bool HasPdp { get; set; }

    public string? UnitName { get; set; }

    public string? StreetNumber { get; set; }

    public string? StreetName { get; set; }

    public string? Suburb { get; set; }

    public string? City { get; set; }

    public string? Province { get; set; }

    public string? PostalCode { get; set; }

    public string? EmployeeNumber { get; set; }

    public string? JobTitle { get; set; }

    public string? Department { get; set; }

    public string? Branch { get; set; }

    public string EmploymentType { get; set; } = "Full-time";

    public string EmploymentStatus { get; set; } = "Active";

    public DateOnly? WorkStartDate { get; set; }

    public string? Manager { get; set; }

    public string? CarTrackDriverId { get; set; }

    public DateTimeOffset CreatedAt { get; set; }

    public string? CreatedByUserId { get; set; }

    public DateTimeOffset UpdatedAt { get; set; }

    public string? UpdatedByUserId { get; set; }
}
